### PR TITLE
API: reimplement FixedWindowIndexer.get_window_bounds

### LINKED ISSUE #36040: BUG: Rolling min_periods not working on groupby object
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [x] (optional) I have confirmed this bug exists on the master branch of pandas.

---

#### Code Sample, a copy-pastable example

```python
df = pd.DataFrame({
    'segment': 'A',
    'data': range(10)
})

df.rolling(5, center=True, min_periods=1).max()

df.groupby('segment').rolling(5, center=True, min_periods=1).max().reset_index(drop=True)
```

#### Problem description

For the DataFrame above, with a single segment 'A', the result of `df.rolling(5, center=True, min_periods=1).max()` should be identical to that of `df.groupby('segment').rolling(5, center=True, min_periods=1).max().reset_index(drop=True)`. Instead, the latter operation has NaNs in the last two positions of the data column.

#### Expected Output

Both operations should return the sequence `[2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0, 9.0, 9.0]`. Instead, `df.groupby('segment').rolling(5, center=True, min_periods=1).max().reset_index(drop=True)` returns  `[2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0, NaN, NaN]`

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : f2ca0a2665b2d169c97de87b8e778dbed86aea07
python           : 3.7.7.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 19.6.0
Version          : Darwin Kernel Version 19.6.0: Thu Jun 18 20:49:00 PDT 2020; root:xnu-6153.141.1~1/RELEASE_X86_64
machine          : x86_64
processor        : i386
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.1.1
numpy            : 1.19.1
pytz             : 2020.1
dateutil         : 2.8.0
pip              : 20.1.1
setuptools       : 47.3.0.post20200616
Cython           : None
pytest           : 6.0.0
hypothesis       : None
sphinx           : 3.1.1
blosc            : None
feather          : None
xlsxwriter       : 1.2.9
lxml.etree       : 4.5.1
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.11.2
IPython          : 7.15.0
pandas_datareader: None
bs4              : None
bottleneck       : None
fsspec           : 0.8.0
fastparquet      : None
gcsfs            : None
matplotlib       : 3.2.1
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : 1.0.0
pytables         : None
pyxlsb           : None
s3fs             : 0.4.2
scipy            : 1.5.2
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : 1.2.0
xlwt             : None
numba            : None

</details>

### PR DESCRIPTION
**Pull request by justinessert**
- [x] closes #36040
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

This PR replaces [36132](https://github.com/pandas-dev/pandas/pull/36132) following @mroeschke's [36567](https://github.com/pandas-dev/pandas/pull/36567).

Originally, my PR was to fix Issue [36040](https://github.com/pandas-dev/pandas/issues/36040). But it appears that Matthew's PR already fixed it! Nonetheless, I still included two things from my previous PR:

1. I added tests to cover the bug outlined in Issue [36040](https://github.com/pandas-dev/pandas/issues/36040).
2. I reimplemented FixedWindowIndexer.get_window_bounds in a way that I believe if a lot more intuitive and simple. Please lmk if you disagree, in which case I could remove this from the PR.

I don't believe any functionality was altered by this PR so I did not include a whatsnew entry

### CODE REVIEW

File: pandas/core/window/indexers.py [hunk 1]
```diff
@@ -78,30 +78,16 @@ def get_window_bounds(
...
+        if center:
+            offset = (self.window_size - 1) // 2
+        else:
+            offset = 0
 
-            start_e_buffer = np.arange(
-                start[-1] + 1, start[-1] + 1 + offset, dtype="int64"
-            )
-            end_e_buffer = np.array([end[-1]] * offset, dtype="int64")
+        end = np.arange(1 + offset, num_values + 1 + offset).astype("int64")
```
Comment (jreback) [inline pandas/core/window/indexers.py:86]: you can add the dtype to the np.arange constructor

File: pandas/tests/window/test_grouper.py [hunk 2]
```diff
@@ -297,6 +297,32 @@ def test_groupby_rolling_center_center(self):
         )
         tm.assert_frame_equal(result, expected)
 
+    @pytest.mark.parametrize("min_periods", [5, 4, 3])
+    def test_groupby_rolling_center_min_periods(self, min_periods):
+        df = pd.DataFrame({"group": ["A"] * 10 + ["B"] * 10, "data": range(20)})
```
Comment (jreback) [inline pandas/tests/window/test_grouper.py:302]: can you add the approrpaite issues numbers as a comment

**Review by jreback — CHANGES_REQUESTED** [review]
looks good @justinessert

**Comment by jreback** [comment]
cc @mroeschke

File: pandas/tests/window/test_grouper.py (hunk 2 above)
Comment (mroeschke) [reply pandas/tests/window/test_grouper.py:302]: e.g `# GH 36040`

File: pandas/core/window/indexers.py (hunk 1 above)
Comment (justinessert) [reply pandas/core/window/indexers.py]: Good call! Added

**Comment by mroeschke** [comment]
Could you also add a whatsnew entry in 1.2?

**Comment by justinessert** [comment]
Looks like there were some failures, but I do not believe those were caused by this PR

**Comment by justinessert** [comment]
> Could you also add a whatsnew entry in 1.2?

@mroeschke I added a whatsnew entry about the reimplementation and linked Issue [36040](https://github.com/pandas-dev/pandas/issues/36040), even though it's not technically fixing that issue (the issue was already fixed by your PR). Please lmk if this is what you intended I add, or if you were looking for something else.

File: doc/source/whatsnew/v1.2.0.rst [hunk 3]
```diff
@@ -444,6 +444,7 @@ Groupby/resample/rolling
 - Bug in :meth:`Rolling.count` returned ``np.nan`` with :class:`pandas.api.indexers.FixedForwardWindowIndexer` as window, ``min_periods=0`` and only missing values in window (:issue:`35579`)
 - Bug where :class:`pandas.core.window.Rolling` produces incorrect window sizes when using a ``PeriodIndex`` (:issue:`34225`)
 - Bug in :meth:`RollingGroupby.count` where a ``ValueError`` was raised when specifying the ``closed`` parameter (:issue:`35869`)
+- Reimplemented :meth:`FixedWindowIndexer.get_window_bounds` in a simpler and more intuitive way (:issue:`36040`).
```
Comment (mroeschke) [inline doc/source/whatsnew/v1.2.0.rst:447]: This entry should describe the bug in #36040

Comment (justinessert) [reply doc/source/whatsnew/v1.2.0.rst]: Ah I see, done

**Review by mroeschke — APPROVED** [review]

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @justinessert

**Pull request merged**